pkg/mic/mic.go

```diff
@@ -470,10 +553,28 @@ func (c *Client) checkIfInUse(checkAssignedID aadpodid.AzureAssignedIdentity, ar
 		if checkID.Spec.Type != aadpodid.UserAssignedMSI {
 			continue
 		}
-		if checkID.Spec.ClientID == id.Spec.ClientID && checkAssignedID.Spec.NodeName == assignedID.Spec.NodeName &&
-			checkAssignedID.Spec.Pod != assignedID.Spec.Pod {
-			return true
+
+		if checkAssignedID.Spec.Pod == assignedID.Spec.Pod {
```

Comment:
If the pod is the same, then shouldn't we check rest before continuing ? or did you want to do != here ?

Reply:
If the pod is the same then there's nothing to check, I think?
The old behavior was returning early if all those things matched.
This just says if it's the same pod then then continue on, which is what the old code did since it would reach the end of the loop rather than return early.